portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.UserLogin;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns current user's email and logged in status. */
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+  private static final String JSON_OUTPUT_TYPE = "application/json";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) 
+      throws IOException {
+    response.setContentType(JSON_OUTPUT_TYPE);
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      UserLogin user = new UserLogin(userEmail, logoutUrl);
+
+      Gson gson = new Gson();
+      String json = gson.toJson(user);
+      response.getWriter().println(json);
+     } else {
+      String urlToRedirectToAfterUserLogsIn = "/";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+      UserLogin user = new UserLogin(loginUrl);
+        
+      Gson gson = new Gson();
+      String json = gson.toJson(user);
+      response.getWriter().println(json);
```

Comment:
optional: this logic appears in both branches of the if/else. You may want to factor it out so that you don't have to have these lines of code twice. you could do this by adding a helper method, or by moving this code to after the if/else block, eg: initialize UserLogin user; before the if, then assign user in the if/else branches, then do this code entirely after the else.

Not required for this CL but going forward it's usually preferable to avoid repeating the same lines of code in multiple places, since that'll make it more readable for your coworkers and easier to change in the future as needed.